File: app/src/main/java/edu/com/softserveinc/bawl/dto/UserHistoryDto.java
package edu.com.softserveinc.bawl.dto;

import java.text.SimpleDateFormat;
import java.util.Date;

public class UserHistoryDto implements Comparable<UserHistoryDto> {

    private String username;
    private String issueName;
    private Date date;
    private String roleName;
    private int statusId;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getIssueName() {
        return issueName;
    }

    public void setIssueName(String issueName) {
        this.issueName = issueName;
    }

    public int getStatusId() {
        return statusId;
    }

    public void setStatusId(int statusId) {
        this.statusId = statusId;
    }

    public String getDate() {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd.mm.yyyy");
        return simpleDateFormat.format(date);
    }

    public void setDate(Date date) {   this.date = date;    }

    public String getRoleName() { return roleName;}

    public void setRoleName(String roleName) {  this.roleName = roleName;  }

    public static int compare(UserHistoryDto o1, UserHistoryDto o2) {
        return o1.date.compareTo(o2.date);
    }

    @Override
    public int compareTo(UserHistoryDto o) {
        return compare(this, o);
    }
}

File: app/src/main/scala/edu/com/softserveinc/bawl/dto/pojo/UserHistoryDTO.scala
package edu.com.softserveinc.bawl.dto.pojo

import java.text.SimpleDateFormat
import java.util.{Comparator, Date}

import scala.beans.BeanProperty

class UserHistoryDTO extends ResponseDTO with Comparable[UserHistoryDTO] with Comparator[UserHistoryDTO] {
   @BeanProperty  var username: String = null
   @BeanProperty  var issueName: String = null
   @BeanProperty  var roleName: String = null
   @BeanProperty  var status: String = ""
   private var date: Date = null
   private var dateStr: String = null

  @Override
  def getDate: String = {
    return dateStr
  }

  def setDate(date : Date) {
    this.date = date
    val simpleDateFormat: SimpleDateFormat = new SimpleDateFormat("dd.mm.yyyy")
    dateStr = simpleDateFormat.format(date)
  }

  def compareTo(o: UserHistoryDTO): Int = {
    return compare(this, o)
  }

  def compare(o1: UserHistoryDTO, o2: UserHistoryDTO): Int = {
    if (o1.date == null || o2.date == null) {
      return -1
    }
    return o1.date.compareTo(o2.date)
  }
}